doc/6/getting-started/.vuejs/standalone/src/App.vue

```diff
@@ -0,0 +1,201 @@
+    // This function return the right formated date depending on the timestamp
+    getDate(timestamp) {
+      const date = new Date(timestamp);
+      return date.toLocaleString().split("GMT")[0];
+    },
+    /* snippet:end */
+    /* snippet:start:6 */
+    // This function will create a message object containing the informations we need to display it
+    getMessage(document) {
+      const message = {
+        // The unique id of the document containing the message
+        _id: document._id,
+        // The text of the message
+        value: document._source.value,
+        // The creation date
+        createdAt: document._source._kuzzle_info.createdAt,
+        // The author name
+        username: document._source.username
+      };
+      return message;
+    },
+    /* snippet:end */
+    /* snippet:start:10 */
+    async sendMessage() {
+      if (this.message === "") return;
+      // Call the create method of the document controller
+      await kuzzle.document.create(
+        "chat",
+        "messages",
+        // Give as parameter the object that will be store in kuzzle
+        {
+          value: this.message,
+          username: this.username
+        }
+      );
+      // Clear the user input
+      this.message = "";
+    },
+    /* snippet:end */
+    /* snippet:start:11 */
+    async subscribeMessages() {
+      // Call the subscribe method of the realtime controller and receive the roomId
+      // Save the id of our subscription (we could need it to unsubscribe)
+      this.roomID = await kuzzle.realtime.subscribe(
+        "chat", // Id of the index
+        "messages", // Id of the collection
+        {}, // Filter
+        // Callback for notifications receive
+        notification => {
+          // Check if the notification interest us (only document creation)
+          if (notification.type !== "document") return;
+          if (notification.action !== "create") return;
+          // Add the new message to our array
+          this.messages = [
+            this.getMessage(notification.result),
+            ...this.messages
+          ];
+        }
+      );
+    },
+    /* snippet:end */
+    /* snippet:start:7 */
+    async fetchMessages() {
+      // Call the search method of the document controller
+      const results = await kuzzle.document.search(
+        "chat", // Name of the index
+        "messages", // Name of the collection
+        { sort: ["_kuzzle_info.createdAt"] }, // Query => Sort the messages by creation date
+        { size: 100 } // Options => get a maximum of 100 messages
+      );
+      // Add each message to our array
+      results.hits.map(hit => {
+        this.messages = [this.getMessage(hit), ...this.messages];
+      });
+    },
+    /* snippet:end */
+    /* snippet:start:2 */
+    async valid() {
+      // Etablish the connection
+      await kuzzle.connect();
+      // Check if 'chat' index exists
+      if (!(await kuzzle.index.exists("chat"))) {
```

Comment:
Nit: unnecessary parenthesis
```suggestion
      if (! await kuzzle.index.exists("chat")) {
```